frontend/src/components/NavItem.tsx

```diff
@@ -0,0 +1,42 @@
+import { NavLink } from "react-router-dom";
+import navStyle from "./Nav.module.css";
+import classNames from "classnames";
+import { useEffect, useState } from "react";
+import { useCookies } from "react-cookie";
+
+export const NavItem = () => {
+  const [cookie] = useCookies(["token"]);
+  const [isCookieAvailable, setIsCookieAvailable] = useState(false);
+
+  useEffect(() => {
+    setIsCookieAvailable((prev) => !prev);
```

Comment:
Dont do this. Be explicit on the logical constraints and functioning of the state variable mirroring your cookie